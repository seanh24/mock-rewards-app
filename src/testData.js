export const testData = [
  {
    sale_id: "02806d8b-b158-4b8c-90db-87e5d3283454",
    customer_id: "083ad411-c2a2-434d-b302-6591fa8389f5",
    product_id: "82eb3ef5-7961-400a-9c96-e0a8dd1469d5",
    sale_price: 73.7,
    sale_time: new Date("Jun 25 2022 03:30:34"),
  },
  {
    sale_id: "50cbd7e7-0b54-46c1-a82c-2d12b809b7a7",
    customer_id: "083ad411-c2a2-434d-b302-6591fa8389f5",
    product_id: "f6880819-7b6c-447a-b20d-08d7d8d0e09c",
    sale_price: 164.3,
    sale_time: new Date("May 12 2022 21:16:32"),
  },
  {
    sale_id: "548905d2-1c3e-4652-a236-c2eedfd4bc06",
    customer_id: "083ad411-c2a2-434d-b302-6591fa8389f5",
    product_id: "3dd6e01c-7eb1-4f5e-a026-6d706dfcdd5c",
    sale_price: 22.67,
    sale_time: new Date("May 18 2022 09:34:32"),
  },
  {
    sale_id: "76dc17de-ae18-482b-b897-f6e0830571da",
    customer_id: "083ad411-c2a2-434d-b302-6591fa8389f5",
    product_id: "f82f0915-93b5-4f49-bef3-14e7c4c511bb",
    sale_price: 121.17,
    sale_time: new Date("Tue May 24 2022 11:43:07"),
  },
  {
    sale_id: "14567a2d-7b25-4b3d-98cc-87e460c942b6",
    customer_id: "083ad411-c2a2-434d-b302-6591fa8389f5",
    product_id: "01a8f565-f0e6-4802-8859-9fb8a798707d",
    sale_price: 188.32,
    sale_time: new Date("Jul 27 2022 16:57:44"),
  },
];
