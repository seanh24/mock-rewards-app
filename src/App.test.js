import { render, screen } from "@testing-library/react";
import CustomerCard from "./components/customerCard";
import { testData } from "./testData";
import "@testing-library/jest-dom";

describe("Customer Card", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<CustomerCard transactions={testData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render a customer card", () => {
    render(<CustomerCard transactions={testData} />);
    const totalRewards = screen.getByTestId("total-rewards");
    const monthlyRewards = screen.getAllByTestId("monthly-reward");
    console.log(totalRewards);
    expect(totalRewards).toBeInTheDocument();
    expect(monthlyRewards).toHaveLength(3);
  });
});
