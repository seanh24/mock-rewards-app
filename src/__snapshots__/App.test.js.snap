// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Card should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="customer-card"
  >
    <h2>
      Customer id: 083ad411-c2a2-434d-b302-6591fa8389f5
    </h2>
    <h3
      data-testid="total-rewards"
    >
      Total Rewards Points: 519
    </h3>
    <h4
      class="customer-card__month"
      data-testid="monthly-reward"
    >
      May: 270 points
    </h4>
    <h4
      class="customer-card__month"
      data-testid="monthly-reward"
    >
      June: 23 points
    </h4>
    <h4
      class="customer-card__month"
      data-testid="monthly-reward"
    >
      July: 226 points
    </h4>
  </section>
</DocumentFragment>
`;
